### ISSUE TITLE
flag provided but not defined seen in share-mnt container

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#25244]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- Create a custom cluster and add a node using the provided command
- See `docker logs share-mnt`

**Result:**
Contains `time="2020-02-02T19:44:05Z" level=fatal msg="flag provided but not defined: -node-name"`

The args is not included in the single string containing the `docker run` but as extra arg which is also used in the actual `share-mnt` container args

This only breaks provisioning on OS's where `share-mnt` is needed to share root. (which most OS don't need)

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.3.5
- Installation option (single install/HA): single

**Comment by sowmyav27** [comment]
**Reproduced the issue on 2.3.5 rancher**

**Steps:**
- Create a custom cluster and add a node using the provided command
- See docker logs share-mnt
- Logs shows

`time="2020-02-10T18:00:48Z" level=fatal msg="flag provided but not defined: -node-name" `

**Verified on master-head - latest - commit id : `3fc6715ca`**

- Create a custom cluster and add a node using the provided command
- See docker logs share-mnt
- Logs DO not show the following message

`fatal msg="flag provided but not defined: -node-name" `
- Logs also show:
```
Found state.json: d55d1cd3486d8858a1ece34e565e0e6739308f2ac3dda1171232174a81d821bf
time="2020-02-10T18:01:23Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/5486/ns/mnt -F -- /var/lib/docker/overlay2/b551644267acd643a58d6b9c8e62020cb01e862d1f79fdaedb10c900dfc9c82c/merged/usr/bin/share-mnt --stage2 /var/lib/kubelet /var/lib/rancher -- norun]" 
+ docker start kubelet
```

**On an upgrade from 2.3.5 to master-head - latest**

Steps:
- Create a custom cluster and add a node using the provided command
- See docker logs share-mnt. Logs shows

`time="2020-02-10T18:00:48Z" level=fatal msg="flag provided but not defined: -node-name" `

- Upgrade rancher to master-head latest
- Logs DO not show the following message

`fatal msg="flag provided but not defined: -node-name" `

Logs show:
```
time="2020-02-10T18:23:19Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/23983/ns/mnt -F -- /var/lib/docker/overlay2/a6b6fada4ba559acb0478f8c7778f57a6ebe200b08cbb368bfdbb7d16918f61b/merged/usr/bin/share-mnt --stage2 /var/lib/kubelet /var/lib/rancher -- norun]" 
+ docker start kubelet
kubelet
+ docker kill --signal=SIGTERM 5c9272ecd073ff8f855b9507b2cc8e5507ae79f3f13ec696a5b10c50f918efee
5c9272ecd073ff8f855b9507b2cc8e5507ae79f3f13ec696a5b10c50f918efee
++ exit 0
```

**Comment by Allen-yan** [comment]
I have the same issue
rancher version:2.3.5
docker version 18.06.3-ce/ 19.03.6
ubuntu 18.04

sudo docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.3.5 --server https://abc.com --token abc123 --etcd --controlplane --worker

**Comment by tico24** [comment]
This issue had been fixed on master, but seems to have reverted.

**Comment by deniseschannon** [comment]
With the changes around share-mnt that were fixed for upgrading, can we re-test this?

**Comment by rancher-max** [comment]
Validated this is still working in master-2401-head `Rancher version 0519da989`. 

Performed the same fresh install and upgrade validations as in Sowmya's comment: https://github.com/rancher/rancher/issues/25244#issuecomment-584274596 and received the same results.

**Comment by nazarewk** [comment]
@mrajashree @superseb is there plan to backport the fix to 2.3 branch? https://github.com/rancher/rancher/pull/25245

**Comment by gizmotronic** [comment]
I'm wondering the same thing, as we're not ready to move to 2.4.x yet.
